prow/apis/prowjobs/v1/types.go

```diff
@@ -286,10 +286,39 @@ type ReporterConfig struct {
 }
 
 type SlackReporterConfig struct {
-	Host              string         `json:"host,omitempty"`
-	Channel           string         `json:"channel,omitempty"`
-	JobStatesToReport []ProwJobState `json:"job_states_to_report,omitempty"`
-	ReportTemplate    string         `json:"report_template,omitempty"`
+	Host              *string         `json:"host,omitempty"`
+	Channel           *string         `json:"channel,omitempty"`
+	JobStatesToReport *[]ProwJobState `json:"job_states_to_report,omitempty"`
+	ReportTemplate    *string         `json:"report_template,omitempty"`
+}
+
+func (src *SlackReporterConfig) ApplyDefault(def *SlackReporterConfig) *SlackReporterConfig {
```

Comment:
Can we add a fuzztest that starts failing if someone adds a new field to `SlackReporterConfig` without adding merge logic, like we have for the decoration utils merging?

Reply:
Added